rs/ic_os/config/src/generate_testnet_config.rs

```diff
@@ -0,0 +1,250 @@
+use anyhow::Result;
+use mac_address::mac_address::FormattedMacAddress;
+use std::net::{Ipv4Addr, Ipv6Addr};
+use std::path::PathBuf;
+use url::Url;
+
+use crate::serialize_and_write_config;
+use crate::types::*;
+
+pub struct GenerateTestnetConfigArgs {
+    // NetworkSettings arguments
```

Comment:
What's the role of this flat struct (why don't we use a nested struct with fields of type NetworkSettings, ICOSSettings, etc.)?